src/main/scala/ai/privado/passes/DBTParserPass.scala

```diff
@@ -0,0 +1,462 @@
+          builder.addEdge(sqlNode, tableNode, EdgeTypes.AST)
+        }
+
+        logger.debug(
+          f">>>>>> DBT Pass Detection: projectName=${projectName}, profileName=${profileName}, modelsDirectoryName=${modelsDirectoryName}, models=${models.length}, dbName=$dbName, dbHost=$dbHost, dbPlatform=$dbPlatform, dbKey: $dbKey, profileFile=$profileFile"
+        )
+      }
+      case Failure(e) => {
+        logger.error(f"error while processing YAML file: ${projectFile}: ${e}")
+        None
+      }
+    }
+  }
+
+  private def transformModelsToSchema(
+    projectName: String,
+    dbPlatform: String,
+    models: Array[java.util.Map[String, Any]]
+  ): DatabaseSchema = {
+    val tables = models.map { table =>
+      val columns = table
+        .get("columns")
+        .asInstanceOf[java.util.List[java.util.Map[String, Any]]]
+        .asScala
+        .map { col =>
+          DatabaseColumn(
+            col.get("name").asInstanceOf[String],
+            col.getOrDefault("description", "").asInstanceOf[String],
+            "",
+            "" // TODO: Match and add sourceId here
+          )
+        }
+        .toList
+
+      DatabaseTable(
+        table.get("name").asInstanceOf[String],
+        table.getOrDefault("description", "").asInstanceOf[String],
+        columns
+      )
+    }.toList
+
+    val dbSchema = DatabaseSchema(kind = "dbt", projectName = projectName, platform = dbPlatform, tables = tables)
+
+    dbSchema
+  }
+
+  private def createDatabaseSinkRule(
+    projectName: String,
+    dbName: String,
+    dbPlatform: String,
+    dbHost: String,
+    schema: DatabaseSchema
+  ) = {
+    val ruleId = f"Storages.DBT.ReadAndWrite.${projectName}"
+
+    val customDatabaseSinkRule = RuleInfo(
+      ruleId,
+      f"${dbName}",
+      "",
+      Array[String]("getdbt.com"),
+      List[String](),
+      false,
+      "",
+      HashMap[String, String](),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SINKS,
+      "storages",
+      Language.DEFAULT,
+      Array[String]()
+    )
+
+    val dbDetails = DatabaseDetails(dbName, dbPlatform, dbHost, "", "", Some(schema))
+
+    ruleCache.setRuleInfo(customDatabaseSinkRule)
+    DatabaseDetailsCache.addDatabaseDetails(dbDetails, ruleId)
+
+    customDatabaseSinkRule
+  }
+
+  private def createSQLNode(builder: DiffGraphBuilder, dbNode: NewDbNode) = {
+    val sqlNode = NewSqlQueryNode().name("DBT").code("DBT").order(0)
+    builder.addNode(sqlNode)
+    builder.addEdge(dbNode, sqlNode, EdgeTypes.AST)
+
+    sqlNode
+  }
+  private def createAndTagDatabaseNode(
+    builder: DiffGraphBuilder,
+    ruleInfo: RuleInfo,
+    projectFile: String,
+    projectName: String,
+    profileFile: String,
+    dbName: String,
+    dbKey: String
+  ) = {
+    val (fileName, (lineNumber, columnNumber, matchedLine)) = dbName match {
+      case "DBT" => (projectFile, findFirstPatternInYAMLFile(projectFile, "name", projectName))
```

Comment:
The string `DBT` is getting repeated multiple times, create a local constant for this, Similarly for other repeated constants